extensions/readme.md

```diff
@@ -0,0 +1,40 @@
+# Jetpack Block Editor Extensions
```

Comment:
Not very important but somehow readmes are usually in allcaps: `README.md` 🤷‍♂️  that's how it seems to be in Jetpack repo as well for other readme files.

Reply:
Good call. Updated in 72f066597ef80174513c585a35a93b2f12d628f4